### ISSUE TITLE
Issue with eslint-config-airbnb-base and `optional chaining` operator

### ISSUE THREAD

**Issue by kayomarz** [airbnb/javascript#2541]
Issue with `eslint-config-airbnb-base` and  [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) operator `?.` `(JavaScript)`

```javascript
// .eslintrc.js
module.exports = {
  env: { es2021: true },
  extends: [ 'airbnb-base' ]
};
```

```javascript
const o = {};
const a = o?.a;
```
For the above code, ESLint reports `Parsing error: Unexpected token` at the optional chaining operator `?.`

ESLint support for **optional chaining** needs `parserOption: {ecmaVersion: 2020}` or above which [eslint automatically sets](https://eslint.org/docs/user-guide/configuring/language-options) when `env: {es2021: true}` is used as above.

However when `airbnb-base` is used, an explicit `parserOption: ecmaVersion: 12` to resolve the above issue as.

```javascript
// .eslintrc.js
module.exports = {
  env: { es2021: true },
  extends: [ 'airbnb-base' ],
  parserOptions: { ecmaVersion: 12 }
};
```

(Note: ecmaVersion: 11 should also work)

The error does not occur without airbnb-base as below:
```javascript
// .eslintrc.js
module.exports = {
  env: { es2021: true }
};
```

Although the fix is simply to explicitly state `ecmaVersion` it can take a while to figure out that the `Parsing error: Unexpected token` is because of airbnb-base.

Other lint errors are not reported in the problem condition making this a bigger issue.

For a better developer experience, this issue could be looked into or documented well.

**Comment by ljharb** [comment]
Does babel-preset-airbnb transform that yet? If so, then https://github.com/airbnb/javascript/blob/b4377fb03089dd7f08955242695860d47f9caab4/packages/eslint-config-airbnb-base/index.js#L13 can be updated.

**Comment by Vivek-04022001** [comment]
Hello, I'm wondering if it's possible for people who are not part of the project to contribute. If the answer is yes, could you please provide guidance on how to do so?

**Comment by ljharb** [comment]
@Vivek-04022001 yes, just look for issues that have a "pull request wanted" label.

**Comment by branzdev** [comment]
Hello,

I'm having the same issue with `eslint-config-airbnb`.

Also, when using the option:

- `"env": { "shared-node-browser": true}`

It does not recognize browser global variables like `document`.

This issue does not happen if I remove `airbnb` from my `extends` configuration.

**Comment by ljharb** [comment]
@branzdev we don't set that env value so i'm not sure why that would affect it. what's your eslint config?

**Comment by martijnhjk** [comment]
I am also having to add `parserOptions: { ecmaVersion: 12 }`, any developments on this topic?

**Comment by ljharb** [comment]
@martijnhjk thats expected pending https://github.com/airbnb/javascript/issues/2541#issuecomment-1014722030

**Comment by martijnhjk** [comment]
@ljharb I am not sure I understand. [That comment](https://github.com/airbnb/javascript/issues/2541#issuecomment-1014722030) was made over a year and a half ago. Will the `ecmaVersion` be updated in the config? Or is overriding `parserOptions` the final recommended practice?
